### PR TITLE
[master < ] Reduce memory consumption on return from python procedures

### PR DESCRIPTION
**Pull request by antoniofilipovic**
[master < Task] PR
- [x] Check, and update documentation if necessary
- [x] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [x] Write a release note here
- [x] Tag someone from docs team in the comments

This PR reduced memory consumption on Python procedures when returning results.

I succeeded to drop memory usage from 8.7GB to 3.5GB by a combination of the following changes:
 - using dedicated memory (object sent to PyModule which was meant to be used for that transformation) for procedures
 - deleting Python objects which were transformed to mgp_value
 - reserving the size of the `result->rows` vector since we know how large `memgraph::utils::pmr::vector<mgp_result_record> rows` of `mgp_result` will we need

All that changes without suffering performance degradation.

Heaptrack before changes:

Screenshot from 2023-05-09 12-56-17

Heaptrack after changes:
Screenshot from 2023-05-10 15-54-24

### CODE REVIEW

**Review by as51340 — APPROVED** [review]
Allez good from my side. Was this before the bug that the memory wasn't passed from EvalContext?

File: src/query/interpreter.hpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ extern const Event FailedQuery;
 namespace memgraph::query {
 
 inline constexpr size_t kExecutionMemoryBlockSize = 1UL * 1024UL * 1024UL;
-inline constexpr size_t kExecutionPoolMaxBlockSize = 2048UL;  // 2 ^ 11
+inline constexpr size_t kExecutionPoolMaxBlockSize = 1024UL;  // 2 ^ 10
```
[the remark this answers is not in the document]
Comment (antoniofilipovic) [reply src/query/interpreter.hpp:54]: To be in sync with memory resources from the interpreter where the limit is also 1024.

[the remark this answers is not in the document]
Comment (antoniofilipovic) [reply src/query/interpreter.hpp:54]: I don't see any benefit in bigger threshold.

**Comment by antoniofilipovic** [comment]
@vpavicic tagging you here for awareness.

Release note can go along the lines: Fixed higher memory consumption when working with Python procedures.

**Comment by antoniofilipovic** [comment]
@vpavicic this was a bug on our side. And we did some memory optimizations to delete some resources once they are transformed from Python to Memgraph's intern C++ values.

> Running Python procedures now consumes less memory.
Works

**Pull request merged**